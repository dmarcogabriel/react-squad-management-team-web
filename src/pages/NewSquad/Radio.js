import React from 'react';
import cn from 'classnames';

const Radio = ({
  className = '',
  label,
  options = [],
  onChange = () => {},
}) => {
  const isSelected = (selected) =>
    selected ? 'border-primary-dark' : 'border-gray-500';

  return (
    <div className={className}>
      <p className="font-bold text-base mb-2">{label}</p>

      <div className="flex justify-between">
        {options.map((option) => (
          <button
            type="button"
            key={option.id}
            className="flex items-center"
            onClick={() => onChange(option)}
          >
            <div
              className={cn(
                'rounded-full p-3 border-1 mr-3 relative',
                isSelected(option.selected)
              )}
            >
              {option.selected && (
                <div className="bg-primary-dark w-5 h-5 absolute m-auto top-0 bottom-0 right-0 left-0 rounded-full" />
              )}
            </div>

            <p
              className={cn(
                'inline text-sm',
                option.selected ? 'text-primary-dark' : 'text-gray-500'
              )}
            >
              {option.label}
            </p>
          </button>
        ))}
      </div>
    </div>
  );
};

export default Radio;
