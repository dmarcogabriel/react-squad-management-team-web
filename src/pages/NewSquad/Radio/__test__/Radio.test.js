import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import userEv from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Radio from '..';

const mockOptions = [
  {
    id: 'm1',
    label: 'Mock Label',
    selected: false,
  },
  {
    id: 'm2',
    label: 'Mock Label',
    selected: true,
  },
];

describe('<Radio />', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Radio options={mockOptions} />, div);
  });

  it('render selected option correctly', () => {
    const { getByTestId } = render(<Radio options={mockOptions} />);

    const radioContainer = getByTestId('radio-m2');

    expect(radioContainer.className).toEqual(
      'rounded-full p-3 border-1 mr-3 relative border-primary-dark'
    );
  });

  it('render not selected option correctly', () => {
    const { getByTestId } = render(<Radio options={mockOptions} />);

    const radioContainer = getByTestId('radio-m1');

    expect(radioContainer.className).toEqual(
      'rounded-full p-3 border-1 mr-3 relative border-gray-500'
    );
  });

  it('render selected option value correctly', () => {
    const { getByTestId } = render(<Radio options={mockOptions} />);

    const radioContainer = getByTestId('value');

    expect(radioContainer.className).toEqual(
      'bg-primary-dark w-5 h-5 absolute m-auto top-0 bottom-0 right-0 left-0 rounded-full'
    );
  });

  it('render selected value label correctly', () => {
    const { getByTestId } = render(<Radio options={mockOptions} />);

    const radioContainer = getByTestId('label-m2');

    expect(radioContainer.className).toEqual(
      'inline text-sm text-primary-dark'
    );
  });

  it('execute click event', () => {
    let result;

    const onChange = (ev) => {
      result = ev;
    };

    const { getByTestId } = render(
      <Radio options={mockOptions} onChange={onChange} />
    );

    const button = getByTestId('button-m1');

    userEv.click(button);

    expect(result).toEqual(mockOptions[0]);
  });

  it('match snapshot', () => {
    expect(renderer.create(<Radio />).toJSON()).toMatchSnapshot();
  });
});
