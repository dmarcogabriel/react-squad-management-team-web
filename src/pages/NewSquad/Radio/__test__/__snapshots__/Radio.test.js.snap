// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> match snapshot 1`] = `
<div
  className=""
>
  <p
    className="font-bold text-base mb-2"
  />
  <div
    className="flex justify-between"
  />
</div>
`;
